### ISSUE TITLE
Archive cert served instead of domains one

### ISSUE THREAD

**Issue by FV-Valraiso** [icing/mod_md#236]
We had a strange behavior on 2 different servers.

The cert served come from the archive folder instead of domains folder.
Therefore the connection is considered insecure because the served cert is expired.
Moreover the behaviour is random, sometimes the valid cert is served.
Other cert renewal were not processed, without any error.

After removing the archive folders of the domain and restart Apache everything goes well.

This happened 2 month ago on a server, and today on another.
And we have no logs about what happened.

A little bit scary

**Comment by icing** [comment]
It sounds more as if there was still an older child process hanging around after a restart and that answered some requests with the old configuration. That would explain the randomness in old/new answers to connections.

Otherwise, mod_md never reads certs from the archived directories, even when domains are missing.

**Comment by cedric-marcone** [comment]
Hello @icing !

I faced the same problem and you were right : a simple Apache restart fixed the issue.
=> It must be caused by a lingering child process.

For the record : 

I tracked the error with the following openssl command :
```sh
openssl s_client -connect host.domain.com:443 -prexit -showcerts
```
=> This command randomly showed a different certificate.

I then greped the offending certificate in `/etc/apache2/md`
```sh
grep xxxx /etc/apache2/md/*/*/*.pem
```
=> It matched the last archived certificate for this domain !

I found this issue opened by @FV-Valraiso with the `archive` keyword.

I ran 
```sh
sudo service apache2 restart
```
=> Fixed

Thank you 👍🏼
